Automattic/woocommerce-payments
# Use individual Gridicons imports

#### Changes proposed in this Pull Request
Most of our Gridicons imports were importing the full module, to reduce our bundle size should import them individually.

- Create a custom ESLint rule to prevent importing the whole Gridicons module again.
- Update `gridicons` TS declaration to meet the new requirement. And move it to a new declarations file in `clien/` as it affects different parts of the project.
- Replace every full Gridicons import with individual ones. Refactoring required code where the icon name was dynamic.
- Update affected test snapshots. And mocking icons to show their name instead of `<Unknown/>`.

#### Testing instructions
- Code changes make sense.
- Tests pass.
- Confirm that they render properly. A good example is a transaction page, where we use different icons in the timeline.

#### Screenshots
<img width="627" alt="Screenshot 2022-10-21 at 10 45 56" src="https://user-images.githubusercontent.com/7670276/197153925-9d6f3f3d-f722-4d5e-89a3-6ff42b3d965c.png">

-------------------


**Post merge**
